app/controllers/checkins_controller.rb

```diff
@@ -12,10 +12,18 @@ def date
     @missing  = @students - @checkins.map(&:student)
   end
 
+  def cohort
+    @students = Student.where(cohort: params[:cohort])
+    p @students
+    @checkins = @students.map(&:checkins)
+    render :students
+  end
+
   private
 
   def setup_nav_data
     @students = Student.order(:name)
     @dates = Checkin.distinct_dates
+    @cohorts = Student.all.map(&:cohort).uniq
```

Comment:
We're using the term 'cohort' throughout the codebase, but it's actually the class name.